ci/pkl-pipelines/common/shared_resources.pkl

```diff
@@ -20,4 +20,26 @@ payCiGitHubResource: PayGitHubResource = new {
         ["uri"] = "https://github.com/alphagov/pay-ci"
         ["branch"] = "master"
     }
+}
+
+function PayDockerHubResource(resource_name: String, repo: String, tag: String): Pipeline.Resource = new {
+    name = resource_name
+    type = "registry-image"
+    icon = "docker"
+    source = new {
+        ["repository"] = repo
+        ["tag"] = tag
+        ["username"] = "((dockerhub-username))"
+        ["password"] = "((dockerhub-password))"
```

Comment:
should this be `((docker-access-token))`?

Reply:
Yeah probably. Have pushed a new commit changning it. The new set-pipeline diff is:

```diff
resources:
  resource detect-secrets-pipeline has been removed:
- icon: github
- name: detect-secrets-pipeline
- source:
-   branch: master
-   paths:
-   - ci/pipelines/detect-secrets.yml
-   uri: https://github.com/alphagov/pay-ci
- type: git

  resource pipeline-source has been added:
+ icon: github
+ name: pipeline-source
+ source:
+   branch: master
+   paths:
+   - ci/pkl-pipelines/detect-secrets.pkl
+   - ci/pkl-pipelines/common/**
+   uri: https://github.com/alphagov/pay-ci
+ type: git

jobs:
  job update-detect-secrets-pipeline has been removed:
- name: update-detect-secrets-pipeline
- plan:
- - get: detect-secrets-pipeline
-   trigger: true
- - file: detect-secrets-pipeline/ci/pipelines/detect-secrets.yml
-   set_pipeline: detect-secrets

  job update-pipeline has been added:
+ name: update-pipeline
+ plan:
+ - get: pipeline-source
+   trigger: true
+ - config:
+     image_resource:
+       name: ""
+       source:
+         repository: governmentdigitalservice/pay-concourse-runner
+         tag: latest
+       type: registry-image
+     inputs:
+     - name: pipeline-source
+     outputs:
+     - name: pipeline-source
+     platform: linux
+     run:
+       args:
+       - -c
+       - pkl eval 'detect-secrets.pkl' > 'detect-secrets.yaml'
+       dir: pipeline-source/ci/pkl-pipelines/
+       path: ash
+   task: generate-yaml-from-pkl
+ - file: pipeline-source/ci/pkl-pipelines/detect-secrets.yaml
+   set_pipeline: detect-secrets-test

```